src/shims/posix/fs.rs

```diff
@@ -1280,30 +1277,38 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
                     immty_from_uint_checked(0u128, c_ushort_layout)?, // d_reclen
                     immty_from_int_checked(file_type, c_uchar_layout)?, // d_type
                 ];
+                let mut name = dir_entry.file_name(); // not a Path as there are no separators!
+                name.push("\0"); // Add a NUL terminator
+                let name_bytes = os_str_to_bytes(&name)?;
+                let name_offset = imms.iter().map(|imm| imm.layout.size.bytes()).sum::<u64>();
```

Comment:
FWIW this violates our usual principle that we only use checked arithmetic... but OTOH the input is fixed and can certainly never overflow a u64, so that seems fine.

Reply:
Fair enough.  Is there a more direct way to get the offset of a field within a struct?  That would actually be convenient so I could calculate it before this to fill in `d_reclen` correctly

Reply:
If it's a struct type one can use `fields.offset` on the `Layout`. Not sure if that is a possibility here though.